### ISSUE TITLE
one `planet` see different `links`/content than other `planet` within same `group`

### ISSUE THREAD

**Issue by david-xlvrs** [urbit/urbit#2578]
**Describe the bug**
I've joined a `group` with my 2 `planets` and realized that the `first planet` see more `links` in a `links collection` than the `second planet`.

**Expected behavior**
Every planet in the `group` should see the same content (e.g. `links` in a `links collection`).

**Desktop (please complete the following information):**
 - MacOS 10.15.4
 - Chrome
 - 0v11.n1t2j.o6f9m.jnh36.ldprs.ksorj.7gpj4.bo1qk.h5a2q.4il4o.nv70v (first planet)
 - 0v11.n1t2j.o6f9m.jnh36.ldprs.ksorj.7gpj4.bo1qk.h5a2q.4il4o.nv70v (second planet)

**Additional context**
The `second planet` was invited to the `group` a few days after the `first planet`.

**Comment by Fang-** [comment]
I suspect the links that are missing are all from the same author(s)? Can the second planet successfully `|hi` them?

**Comment by david-xlvrs** [comment]
Hmm, weird — I can't `|hi` even to `~dopzod` or `~zod`. Last message in `~/~dopzod/urbit-help` is from 3 hours ago.

**Comment by david-xlvrs** [comment]
> I suspect the links that are missing are all from the same author(s)? Can the second planet successfully `|hi` them?

Okay, I'm back with more info. I've investigated that 2 `planets` in the `group` which were invited at the same time (second cohort, ~1 day after the `group` was created) can't see some of the content in the `links collections`. The `links` which are missing were added by 2 different `planets` and I can `|hi` them successfuly. Also, probably all `planets` from the first cohort of invites can see all of the content in the `links collections`.

**Comment by Fang-** [comment]
> The links which are missing were added by 2 different planets and I can |hi them successfuly.

But can the planets that can't see their content `|hi` them, too?

**Comment by david-xlvrs** [comment]
> > The links which are missing were added by 2 different planets and I can |hi them successfuly.
> 
> But can the planets that can't see their content `|hi` them, too?

Yes, I have one of these planets which can't see the content and I can `|hi` them.

**Comment by david-xlvrs** [comment]
Also, at least one of these planets which content some other planets can't see are running on a server 24/7.

**Comment by Fang-** [comment]
Okay, sorry, I should've been more clear with these questions. Background here is that links works peer-to-peer, so _if you want to see the links someone posted, you have to be able to talk to them directly_, regardless of other people in the group. Being online 24/7 is good, but sometimes two planets can't talk to each other despite both being online.

You say two planets are missing content, not seeing things that you can see. Let's call these ~A and ~B.

You also say that the content ~A and ~B are missing is from two different authors. Let's call these authors ~C and ~D.

My question: can ~A `|hi ~C`? Can ~A `|hi ~D`? Can ~B `|hi ~C`? Can ~B `|hi ~D`?

(Might also be good to know if ~C can `|hi ~A` etc, but imagine that's less relevant here.)

**Comment by david-xlvrs** [comment]
We did some tests and results are:

Planet `~A` can `|hi` planet  `~C` and `~D` successfully
Planet `~B` can `|hi` planet `~C` and  `~D` successfully
Planet `~C` can `|hi` planet `~A` and `~B` successfully

I think we can invite you to the `group` if it helps.

**Comment by Fang-** [comment]
If you can, that'd be great! Maybe then we can chat to try and debug this further, too, if needed.

**Comment by urcades** [comment]
@Fang- you've been added to the group

**Comment by Fang-** [comment]
Alright I looked into this. I see it happening for this group, and others as well.

Running `:link-listen-hook +dbug [%state 'retry-timers']` shows that I _am_ still trying to reach out to the ships whose links I'm missing, but the presence of `retry-timers` indicates that my subscription requests keep getting rejected.

Trying to see if my ship was doing the same to others, I added some printfs to link-proxy-hook. And indeed, it is rejecting subscription requests. Why? Because my ship doesn't know the subscribing ship's part of the group, and so rejects it with "permission denied".

Went and confirmed with someone who was in @urcades' group before me. They see 8 members, but I'm not there. I joined the group more recently, and I see 11 members.

So, link is working fine, but group member propagation is borked, and link depends on that...
